preprocess.py

```diff
@@ -1,3 +1,65 @@
 '''
-moves data from `data/raw` to `data/processed`
-'''
\ No newline at end of file
+algorithm to pull data from reddit,
+determine whether the poster is an asshole based on the commenters annotattions,
+and output to file
+(maybe link with EC2?)
+'''
+import praw
+import os
+
+from praw.models import MoreComments
+
+import config
+from pathlib import Path
+import csv
+
+
+def is_asshole(submission: praw.models.reddit.submission.Submission):
+    """
+    iterate through the comments to find signals for asshole or not
+    NTA = not the asshole
+    YTA = you're the asshole
+    :param submission: a praw.models.reddit.submission.Submission object
+    :return: True if more comments have 'YTA' than 'NTA'
+    """
+    votes_for_NTA = 0
+    votes_for_YTA = 0
+
+    # This submission’s comment forest contains a number of MoreComments objects.
+    # These objects represent the “load more comments”,
+    # and “continue this thread” links encountered on the website.
+    # To remove them, we can call replace_more with a limit of 0
+    # to remove ALL MoreComments
+    submission.comments.replace_more(limit=0)
+    for comment in submission.comments:
+        text = comment.body
+        if 'NTA' in text:
```

Comment:
I can imagine that we may want to encapsulate the "NTA" and "YTA" decisions as functions

Reply:
Good idea. I separated it out, but it got a little messier because of a null check. Let me know what you think